components/quick_input/quick_input.tsx

```diff
@@ -11,65 +11,75 @@ import OverlayTrigger from 'components/overlay_trigger';
+     */
+    onClear?: () => any;
+
+    /**
+     * ClassName for the clear button container
+     */
+    clearClassName?: string;
+
+    /**
+     * Position in which the tooltip will be displayed
+     */
+    tooltipPosition?: 'top' | 'bottom';
+
+    /**
+     * Callback to handle the change event of the input
+     */
+    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+
+    /**
+     * When true, and an onClear callback is defined, show an X on the input field even if
+     * the input is empty.
+     */
+    clearableWithoutValue?: boolean;
+
+    maxLength?: number;
+    className?: string;
+    placeholder?: string | { id: any; defaultMessage: string };
+    autoFocus?: boolean;
+    type?: string;
+    id?: string;
+    onInput?: () => void;
+}
+
 // A component that can be used to make controlled inputs that function properly in certain
 // environments (ie. IE11) where typing quickly would sometimes miss inputs
-export default class QuickInput extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * Whether to delay updating the value of the textbox from props. Should only be used
-         * on textboxes that to properly compose CJK characters as the user types.
-         */
-        delayInputUpdate: PropTypes.bool,
-
-        /**
-         * An optional React component that will be used instead of an HTML input when rendering
-         */
-        inputComponent: PropTypes.elementType,
-
-        /**
-         * The string value displayed in this input
-         */
-        value: PropTypes.string.isRequired,
-
-        /**
-         * When true, and an onClear callback is defined, show an X on the input field that clears
-         * the input when clicked.
-         */
-        clearable: PropTypes.bool,
-
-        /**
-         * The optional tooltip text to display on the X shown when clearable. Pass a components
-         * such as FormattedMessage to localize.
-         */
-        clearableTooltipText: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
-
-        /**
-         * Callback to clear the input value, and used in tandem with the clearable prop above.
-         */
-        onClear: PropTypes.func,
-
-        /**
-         * ClassName for the clear button container
-         */
-        clearClassName: PropTypes.string,
-
-        /**
-         * Position in which the tooltip will be displayed
-         */
-        tooltipPosition: PropTypes.oneOf(['top', 'bottom']),
-
-        /**
-         * Callback to handle the change event of the input
-         */
-        onChange: PropTypes.func,
-
-        /**
-         * When true, and an onClear callback is defined, show an X on the input field even if
-         * the input is empty.
-         */
-        clearableWithoutValue: PropTypes.bool,
-    };
+export default class QuickInput extends React.PureComponent<Props> {
+    private input: any;
```

Comment:
`input` should be of type `HTMLInputElement`
```suggestion
    private input: HTMLInputElement;
```

Reply:
Yes, it was my first try but I had 2 issues with that : it has to be undefined or defined in constructor (but with which value?) and there is a file that performs a cast to a `AutosizeTextarea`, so i simplified to any. Please tell me if b29e36ef80f1 is on the good path

Reply:
Yeah I think that works for now.